### PR TITLE
feat(@aws-amplify/ui-components): AmplifyAuthFields accept type for default fields

### PR DESCRIPTION
**Pull request by mayt**
_Issue #6689_

_Description of changes:_
Set the type props when using `amplify-form-field`. And fix the "amplify-auth-fields spec: Render logic -> should render custom type fields" test so it actually renders the form components

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-ui-components/src/components/amplify-auth-fields/amplify-auth-fields.spec.ts [hunk 1]
```diff
@@ -48,10 +48,14 @@ describe('amplify-auth-fields spec:', () => {
       const usernameField = ['username', 'password'];
 
       const page = await newSpecPage({
-        components: [AmplifyAuthFields],
-        html: `<amplify-auth-fields formFields=${usernameField}></amplify-auth-fields>`
+        components: [AmplifyAuthFields]
       });
 
+      let component = page.doc.createElement("amplify-auth-fields");
```
Comment (wlee221) [inline packages/amplify-ui-components/src/components/amplify-auth-fields/amplify-auth-fields.spec.ts:54]:
```suggestion
      const component = page.doc.createElement("amplify-auth-fields");
```

File: packages/amplify-ui-components/src/components/amplify-auth-fields/amplify-auth-fields.spec.ts [hunk 2]
```diff
@@ -63,5 +67,20 @@ describe('amplify-auth-fields spec:', () => {
...
     });
+
+    it('should render custom type fields', async () => {
+      const customField = [{'label': 'date', 'type': 'date'}];
+
+      const page = await newSpecPage({
+        components: [AmplifyAuthFields]
+      });
+
+      let component = page.doc.createElement("amplify-auth-fields");
```
Comment (wlee221) [inline packages/amplify-ui-components/src/components/amplify-auth-fields/amplify-auth-fields.spec.ts:78]:
```suggestion
      const component = page.doc.createElement("amplify-auth-fields");
```

**Review by wlee221 — CHANGES_REQUESTED** [review]
Hello @mayt, sorry about the delay in reviewing this. This LGTM if we just fix the eslint change requests above.

**Comment by mayt** [comment]
I have merged in the suggestions.

**Review by wlee221 — APPROVED** [review]
LGTM, thanks @mayt for the contribution.

**Pull request merged**